cmake/appleclang-toolchain.cmake

```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This toolchain file is not meant to be used directly,
+# but to be invoked by CMake preset and GitHub CI.
+#
+# This toolchain file configures for apple clang family of compiler.
+# Note this is different from LLVM toolchain.
+#
+# BEMAN_BUILDSYS_SANITIZER:
+# This optional CMake parameter is not meant for public use and is subject to
+# change.
+# Possible values:
+# - MaxSan: configures clang and clang++ to use all available non-conflicting
+#           sanitizers. Note that apple clang does not support leak sanitizer.
+# - TSan:   configures clang and clang++ to enable the use of thread sanitizer.
+
+set(CMAKE_C_COMPILER clang)
+set(CMAKE_CXX_COMPILER clang++)
+
+if(BEMAN_BUILDSYS_SANITIZER STREQUAL "MaxSan")
+    set(SANITIZER_FLAGS
+        "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=undefined"
+    )
+elseif(BEMAN_BUILDSYS_SANITIZER STREQUAL "TSan")
+    set(SANITIZER_FLAGS "-fsanitize=thread")
+endif()
+
+set(CMAKE_C_FLAGS_DEBUG_INIT "${SANITIZER_FLAGS}")
+set(CMAKE_CXX_FLAGS_DEBUG_INIT "${SANITIZER_FLAGS}")
+
+set(RELEASE_FLAG "-O3 ${SANITIZER_FLAGS}")
```

Comment:
I think this variable should be prefixed with `BEMAN_BUILDSYS_` to avoid conflicts with project variables.

Reply:
This is only meant to be a local variable and not to be exported. I will see if I can make it so that it's treated like a local variable.

Reply:
I don't think this variable is exported out. Being a toolchain file, this shouldn't overwrite anything. `BEMAN_BUILDSYS_` implies this variable to be externally facing.

Reply:
Oops, you're right. I didn't realize that variables set in the toolchain don't propagate.